storage/innobase/srv/srv0start.cc

```diff
@@ -1996,6 +1996,58 @@ static lsn_t srv_prepare_to_delete_redo_log_files(ulint n_files) {
   return (flushed_lsn);
 }
 
+/** Enable encryption of system tablespace if requested. At
+startup load the encryption information from first datafile
+to tablespace object
+@return DB_SUCCESS on succes, others on failure */
+static dberr_t srv_sys_enable_encryption(bool create_new_db) {
+  fil_space_t *space = fil_space_get(TRX_SYS_SPACE);
+  dberr_t err = DB_SUCCESS;
+
```

Comment:
Original patch used to have
```
  // Fail startup if sys space is encrypted with crypt_data v1
  // This should only happen on upgrade
  if (srv_sys_space.keyring_encryption_info.page0_has_crypt_data &&
      srv_sys_space.keyring_encryption_info.type != CRYPT_SCHEME_UNENCRYPTED &&
      srv_sys_space.keyring_encryption_info.private_version == 1) {
    ib::error(ER_UPGRADE_KEYRING_V1_ENCRYPTION);
    return (DB_ERROR);
  }
```
here

Reply:
Hmm, this should be brought back again. If there is crypt_data old format in system tablespace, we disallow upgrades.

Reply:
Done!